plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/data/LayoutMetaDataService.java

```diff
@@ -29,11 +40,50 @@
  * @kieler.rating yellow 2012-10-09 review KI-25 by chsch, bdu
  * @author msp
  */
-public class LayoutMetaDataService {
+public final class LayoutMetaDataService {
 
     /** the layout data service instance, which is created lazily. */
     private static LayoutMetaDataService instance;
 
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    private LayoutMetaDataService() {
+        
+        // every class that implements IDataObject 
+        // or is used for internal properties must be registered here
```

Comment:
It's a bit strange that the service needs to know which option types will be registered to it. This should rather be the responsibility of the ILayoutMetaDataProviders.

Can we generate this code through the MetaData language?